server/src/main/java/io/crate/execution/dml/Indexer.java

```diff
@@ -661,15 +693,46 @@ public ParsedDocument index(IndexItem item) throws IOException {
         }
     }
 
+    /**
+     *
+     * RETURNING can refer to multiple non-deterministic columns,
+     * generated and defaults and their combinations, e.g., gen_col + def_col.
+     *
+     * We need to ensure that we use already computed values for such references,
+     * this this must be called after {@link #index(IndexItem)} have been called.
+     *
+     */
```

Comment:
```suggestion
```

Method docs should be primarily for API consumers. This is addressing the implementation. I think we should move this inside the method body but see also my comment below. This might even become unnecessary if we change the Synthetic to an Input